runner/execer/os/os_execer.go

```diff
@@ -89,21 +90,35 @@ func (p *osProcess) monitorMem(memCap execer.Memory, stat stats.StatsReceiver) {
 	reportThresholds := []float64{0, .25, .5, .75, .85, .9, .93, .95, .96, .97, .98, .99, 1}
 	memTicker := time.NewTicker(10 * time.Millisecond)
 	defer memTicker.Stop()
+	// Clean up after ourselves
+	pgid, err := syscall.Getpgid(pid)
+	if err != nil {
+		log.Errorf("Error finding pgid of pid %d, %v", pid, err)
+	} else {
+		defer cleanupProcs(pgid)
+	}
 	log.Infof("Monitoring memory for pid=%d", pid)
 	for {
 		select {
 		case <-memTicker.C:
 			p.mutex.Lock()
+			// Process is complete
 			if p.result != nil {
 				p.mutex.Unlock()
 				log.Infof("Finished monitoring memory for pid=%d", pid)
 				return
 			}
 			mem, _ := memUsage(pid)
 			stat.Gauge("memory").Update(int64(mem))
+			// Aborting process, above memCap
 			if mem >= memCap {
-				msg := fmt.Sprintf("Cmd exceeded MemoryCap, aborting: %d > %d (%v)", mem, memCap, p.cmd.Args)
-				log.Info(msg)
+				msg := fmt.Sprintf("Cmd exceeded MemoryCap, aborting %d: %d > %d (%v)", pid, mem, memCap, p.cmd.Args)
```

Comment:
In loglens:
<img width="1253" alt="screen shot 2017-05-10 at 11 36 16 am" src="https://cloud.githubusercontent.com/assets/3813401/25907253/f5555eb4-3574-11e7-952e-c02e532d3428.png">
In the worker:
<img width="633" alt="screen shot 2017-05-10 at 11 38 47 am" src="https://cloud.githubusercontent.com/assets/3813401/25907395/5999ce50-3575-11e7-8e9a-f962d8b497c4.png">